File: backend/src/main/java/com/agriconnect/service/ProductService.java
package com.agriconnect.service;

import com.agriconnect.constant.ProductStatus;
import com.agriconnect.dto.request.ProductRequest;
import com.agriconnect.dto.response.PageResponse;
import com.agriconnect.dto.response.ProductResponse;
import org.springframework.data.domain.Pageable;

public interface ProductService {

    PageResponse<ProductResponse> listApproved(String search, String category, String location, Boolean available, Boolean organic, Pageable pageable);

    PageResponse<ProductResponse> listMine(Pageable pageable);

    ProductResponse getById(Long id);

    ProductResponse create(ProductRequest request);

    ProductResponse update(Long id, ProductRequest request);

    void delete(Long id);

    PageResponse<ProductResponse> listForAdmin(ProductStatus status, String search, Pageable pageable);

    ProductResponse updateApprovalStatus(Long productId, ProductStatus status);

    ProductResponse updateQuantity(Long productId, java.math.BigDecimal quantity);

    long countByApprovalStatus(ProductStatus status);
}

File: frontend/src/services/productService.js
import api from './api';

export const productService = {
  listApproved: (params) => api.get('/products', { params }),
  getById: (id) => api.get(`/products/${id}`),
  listMine: (params) => api.get('/products/mine', { params }),
  create: (data) => api.post('/products', data),
  update: (id, data) => api.put(`/products/${id}`, data),
  delete: (id) => api.delete(`/products/${id}`),
  updateQuantity: (id, data) => api.patch(`/products/${id}/quantity`, data),
};

export const productOrderService = {
  placeOrder: (data) => api.post('/product-orders', data),
  listMyOrders: (params) => api.get('/product-orders/mine', { params }),
  listIncomingOrders: (params) => api.get('/product-orders/incoming', { params }),
  getById: (id) => api.get(`/product-orders/${id}`),
  updateStatus: (id, data) => api.patch(`/product-orders/${id}/status`, data),
};
